src/server/pkg/deploy/cmds/cmds.go

```diff
@@ -629,6 +629,9 @@ If <object store backend> is \"s3\", then the arguments are:
 		"(required) Backend providing an object-storage API to pachyderm. One of: "+
 			"s3, gcs, or azure-blob.")
 	deployCustom.Flags().BoolVar(&isS3V2, "isS3V2", false, "Enable S3V2 client")
+	if isS3V2 {
```

Comment:
So this check has to go up in the `deployCustom` `Run` callback above - the code at this level is run every time `pachctl` runs, but before arg-parsing, so this variable will always be `false`.  Just cutting-pasting this to the start of the `Run` callback should be good.

Reply:
Interesting. I thought this was under the standardDeployCmds, so it would only be run when someone uses the deploy command. But I guess any time sone types `pachctl deploy --help` this will also run to display all the help.

Fixed it.